framework/elements/tree/function_to_tree.py

```diff
@@ -1,74 +1,72 @@
+from search_space.populate_search_space import PopulateSearchSpace
 
 
 class StringToTree:
 
-    parenMarker = 0  # Helps indicates if a substring is surrounded by open and closed parentheses
+    paren_marker = 0  # Helps indicates if a substring is surrounded by open and closed parentheses
 
-    parenIndex = -1  # Marks positions of parenthesis in the function string
+    paren_index = -1  # Marks positions of parenthesis in the function string
 
-    cropIndex = [0, 0]  # Indicates the positions of two corresponding open and closed parentheses
+    crop_index = [0, 0]  # Indicates the positions of two corresponding open and closed parentheses
 
-    firstTreeList = []  # Inital representation of function tree
+    first_tree_list = []  # Inital representation of function tree
 
-    subLists = []  # Stores sublist for recursion
+    sub_lists = []  # Stores sublist for recursion
 
     operands = {"prod": "*", "sum": "+", "diff": "-", "quot": "/", "mod": "%"}  # Acts as a operator bank 
 
-    finalTreeList = []  # Final (Level Traversed) representation of function tree 
+    final_tree_list = []  # Final (Level Traversed) representation of function tree 
 
-    kidslist = []  # Stores kids of each level traversed node
+    kids_list = []  # Stores kids of each level traversed node
 
-    '''String to tree:
+    '''
+    String to tree:
         This class uses three functions to convert from a function to a usable
-        tree for evolf based on the function paramenters.
-
-        StringToTree1: This turns a function to a list that contains nodes 
+    tree for evolf based on the function paramenters.To make use of the class 
+    use the function string_to_tree which combines all three of the functions in 
+    question.Description of questions seen below.
+    '''
+    @classmethod
+    def function_to_level_list(cls, string, level=0):
+        '''
+        function_to_level_list: 
+            This turns a function to a list that contains nodes 
         represented as "[node function, level, children]" where the level 
         is a marker of how deep a node is in the function
+        '''
 
-        StringToTree2: This turns the output from the above function into a 
-        similar function with null nodes included
-
-        StringToTree3: This turns the output from the above function above 
-        into a level trvaversed list of function names.
-
-        To make use of the class use the function StringToTree which combines
-        all three of the functions above.
-              '''
-    @classmethod
-    def stringToTree1(cls, string, level=0):
-
-        cls.subLists = []
+        cls.sub_lists = []
         string = string.replace(" ", "")
         if ')' and ',' not in string:
 
-            cls.firstTreeList.append([string, level, ""])
+            cls.first_tree_list.append([string, level, ""])
             return string
 
         else:
 
             for i in string:
```

Comment:
Do you mean, "for character in string:"? Please use better variables and iterator names.